cfme/tests/middleware/test_middleware_deployment.py

```diff
@@ -1,33 +1,48 @@
-from random import sample
-
 import pytest
+from cfme.middleware import get_random_list
 from cfme.middleware.deployment import MiddlewareDeployment
+from cfme.middleware.server import MiddlewareServer
 from utils import testgen
 from utils.version import current_version
 
 pytestmark = [
-    pytest.mark.usefixtures('single_middleware_provider'),
+    pytest.mark.usefixtures('setup_provider'),
```

Comment:
@hhovsepy as @mfalesni mentioned `provider = get_crud(get_provider_key(provider_name))` we can use this method to create provider object.
When doing operations like `edit`, `delete`, `disable`. at the end of the test we need to keep resource to original state [or] in that case we may do recreation of `provider`(for that test alone).